playbooks/roles/prometheus/tasks/grafana.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: Grafana – add apt repository
+  apt_repository:
+    repo: deb https://packagecloud.io/grafana/stable/debian/ stretch main
+    filename: grafana
+
+- name: Grafana – add apt key
+  apt_key:
+    url: https://packagecloud.io/gpg.key
+
+- name: Grafana – install Debian package
+  apt:
+    name: grafana
+    update_cache: yes
+
+- name: Grafana – only listen on localhost
+  lineinfile:
+    path: /etc/grafana/grafana.ini
+    regexp: "http_addr.*="
+    line: "http_addr = 127.0.0.1"
+
+- name: Grafana – start server
+  service:
+    name: grafana-server
+    state: started
+    enabled: yes
+
+# This task does not work on Python 3, and is commented out for this reason.
```

Comment:
:(

I haven't looked into it much, but maybe we can use https://github.com/cloudalchemy/ansible-grafana/blob/master/tasks/datasources.yml ?

Reply:
@UmanShahzad I will look into this when setting up Grafana, which we have a separate ticket for.

I first need to figure out whether Python 2 is required on the target or on the host running Ansible.  If it's the former, we could use Python 2 on the target just for these tasks, which should be possible by adding

    vars:
      ansible_python_interpreter: /usr/bin/python

If it's the latter, we will have to keep using Python 2 virtualenvs for running Ansible, but this shouldn't really be a big deal – we just need to document it.

Alternatively, we may even be able to fix the bug and submit a patch upstream.

Reply:
Alright, let's leave this to the Grafana task.